### PR TITLE
[MOS-561] Minimal documentation on how update works

### PR DESCRIPTION
**Pull request by pholat**
Added minimal documentation on upate:
- how it works
- what is the arhitecture behind

**Description**

Please see: https://github.com/mudita/MuditaOS/blob/be19691e5cff9a753bf62920277706b7b5c43081/doc/boot_and_update.md

**Your checklist for this pull request**

Make sure that this PR:
- [x] Has documentation updated

### CODE REVIEW

File: doc/boot_and_update.md [hunk 1]
```diff
@@ -1,5 +1,132 @@
 # How to boot Mudita Pure and create a storage partition
 
+## Generall upadte architecture
```
Comment (Maciej-Mudita) [inline doc/boot_and_update.md:3]: update

File: doc/boot_and_update.md [hunk 2]
```diff
@@ -1,5 +1,132 @@
...
+
+There are two mayor ways to update the OS:
```
Comment (Maciej-Mudita) [inline doc/boot_and_update.md:5]: major

File: doc/boot_and_update.md (hunk 1 above)
Comment (pholat) [reply doc/boot_and_update.md:3]: tx I'm checking my poor writing skills just now :) (And requesting @SP2FET to help me with that as he always proof reads my documentation :D )

**Review by Maciej-Mudita — APPROVED** [review]

File: doc/boot_and_update.md (hunk 2 above)
Comment (SP2FET) [inline doc/boot_and_update.md:5]:
```suggestion
There are two major ways to update the OS:
```

File: doc/boot_and_update.md [hunk 3]
```diff
@@ -1,5 +1,132 @@
...
+- via update utility
+- development only MMC access
+
+Generall applications interaction diagram.
```
Comment (SP2FET) [inline doc/boot_and_update.md:9]:
```suggestion
General applications interaction diagram:
```

File: doc/boot_and_update.md [hunk 4]
```diff
@@ -1,5 +1,132 @@
...
+```
+                              +------------+
+       +_____ [reboot] _______| MuditaOS   |
+       ↓                      | (boot.bin) |
+------------+                 +------------+
+|bootloader  |                        ↑
+|ecoboot.bin |--[load and boot]------→◇
++------------+                        ↓
+    ↑  ↓                            +--------------+
+    |  |-----[upadte flag]---------→| Updater      |
```
Comment (SP2FET) [inline doc/boot_and_update.md:20]:
```suggestion
    |  |-----[update flag]---------→| Updater      |
```

File: doc/boot_and_update.md [hunk 5]
```diff
@@ -1,5 +1,132 @@
...
+    - reboot with clean boot flag
+        - reboots with clean boot flag
+- OS, from update perspective
+    - accepts tar update package
+    - reboots the OS with SNVS flag set to update
+    - **nothing more**
+
+**Important**
+The updater is by design:
+- as OS agnostic as possible.
```
Comment (SP2FET) [inline doc/boot_and_update.md:56]:
```suggestion
- as OS-agnostic as possible
```

File: doc/boot_and_update.md [hunk 6]
```diff
@@ -1,5 +1,132 @@
...
+- as simple as possible
+- does only one procedure at the time with fail fast aproach
```
Comment (SP2FET) [inline doc/boot_and_update.md:58]:
```suggestion
- does only one procedure at a time with fail-fast approach
```

File: doc/boot_and_update.md [hunk 7]
```diff
@@ -1,5 +1,132 @@
...
+- procedural software with:
+    - short upadte trace
```
Comment (SP2FET) [inline doc/boot_and_update.md:60]:
```suggestion
    - short update trace
```

File: doc/boot_and_update.md [hunk 8]
```diff
@@ -1,5 +1,132 @@
...
+
+### TL;DR; What updater provides:
+
+1. update
+2. restore: from last update to previous system version
+3. factory reset: it only cleans data, does not restore factory OS
+
+## Why is the updater a separate updater application
+
+To assert that the process fully separated from the OS and as abstract from the
```
Comment (SP2FET) [inline doc/boot_and_update.md:93]:
```suggestion
To assert that the process is fully separated from the OS and as abstract from the 
```

File: doc/boot_and_update.md [hunk 9]
```diff
@@ -1,5 +1,132 @@
...
+2. restore: from last update to previous system version
+3. factory reset: it only cleans data, does not restore factory OS
+
+## Why is the updater a separate updater application
+
+To assert that the process fully separated from the OS and as abstract from the 
+hardware as possible and finally as simple the update process was designed as 
+above.
+
+# Quick and dirty how to debug update issues:
```
Comment (SP2FET) [inline doc/boot_and_update.md:97]:
```suggestion
# Quick and dirty way of how to debug update issues:
```

File: doc/boot_and_update.md [hunk 10]
```diff
@@ -1,5 +1,132 @@
...
+1. download package affected
```
Comment (SP2FET) [inline doc/boot_and_update.md:98]:
```suggestion
1. download update package
```

File: doc/boot_and_update.md [hunk 11]
```diff
@@ -1,5 +1,132 @@
...
+    *OR*
+    load updater.bin
+    *OR* 
+    from the example: check why there is no checksum in the first place
+        - check if file with checksum exists
+        - check if it has checksum in it
+    Then: 
+    - it there is no checksum **FIXIT** on the CI/locally (for local builds)
+    - if there was checksum, and updater failed **FIXIT** in the updater
+5. status always in 3:/updater.log
```
Comment (SP2FET) [inline doc/boot_and_update.md:127]:
```suggestion
5. status is always in 3:/updater.log
```

**Review by SP2FET — APPROVED** [review]

**Pull request merged**